agent/nmap_agent.py

```diff
@@ -372,6 +372,19 @@ def _emit_fingerprints(
                 else:
                     raise ValueError(f"Incorrect ip version {version}")
 
+                if host.get("os") is not None:
+                    if host.get("os").get("osmatch", {}) is not None:
+                        logger.debug("sending results to selector %s", selector)
+
+                        os = host.get("os").get("osmatch", {})[0]
+                        fingerprint_data = {
+                            "host": host.get("address", {}).get("@addr"),
+                            "library_type": "OS",
+                            "library_name": os.get("osclass").get("@osfamily"),
+                            "library_version": os.get("osclass").get("@osgen"),
+                        }
```

Comment:
i think you should add other fields:
![image](https://github.com/user-attachments/assets/b35883b7-b76b-4c2d-af67-3df452273c44)

Reply:
fields are optional

Reply:
Still it might be interesting to have them.

Reply:
this fingerprint model is designed for a service fingerprint and those fields (like port protocol etc...) can't have them for a OS